src/channel/command_pool.rs

```diff
@@ -0,0 +1,91 @@
+use crate::APP_CONF;
+use rayon::{ThreadPool, ThreadPoolBuilder};
+use std::sync::{Arc, Mutex};
+
+lazy_static! {
+    pub static ref COMMAND_POOL: CommandPool = CommandPool::new();
+}
+
+pub struct CommandPool {
+    pub thread_pool: Arc<Mutex<ThreadPool>>,
+}
+
+impl CommandPool {
+    pub fn new() -> Self {
+        let num_threads = APP_CONF.channel.command_pool_num_threads;
+        debug!("initializing command pool, num threads: {}", num_threads);
+        Self {
+            thread_pool: Arc::new(Mutex::new(
+                ThreadPoolBuilder::new()
+                    .num_threads(num_threads)
+                    .build()
+                    .unwrap(),
+            )),
+        }
+    }
+
+    pub fn enqueue<'a, F>(&self, func: F)
+    where
+        F: FnOnce() + Send + 'a,
+    {
+        self.thread_pool.lock().unwrap().scope(|s| {
+            s.spawn(move |_| func());
+        });
```

Comment:
`scope` is executing tasks within scoped thread pool, the thing is that this operation blocks the main thread. I need to rework that.